experimental/vjn/yarpl/include/reactivestreams/subscriber.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+
+#include "type_traits.h"
+
+namespace reactivestreams {
+
+template <typename T>
+class Subscription;
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+
+  virtual void on_next(const T&) {}
```

Comment:
nit: can't say that I like the naming conventions. It looks like naming conventions from the standard library, but the class name starts with a capital letter, so this feels like a hybrid ;)